### ISSUE TITLE
Add multiple topic subscription

### ISSUE THREAD

**Issue by garthvh** [brianshea2/meshmap.net#21]
Hi @brianshea2,

We have completed some updates to the packet filtering on the meshtastic mqtt server so that only positions with a precision form 10-16 will be sent out globally over mqtt, new firmware will default to 13.

MeshMap was collateral damage when the global root subscription was disabled, and there is no plan to re-enable the global subscription but it should be pretty easy to get meshmap back up and running by looping through the topics for a region and subscribing to those.

Your patience and respect for users privacy on this has been appreciated.

**Comment by brianshea2** [comment]
Hi @garthvh! I was going to reach out to you on discord, thank you for beating me to it :)

Glad to hear the privacy focused improvements have been made. I'm happy to re-enable MeshMap by subscribing to each of the official region-based root topics. Confirming, that's what's wanted? That will mean custom root topics (as is popular in the USA with adding the state, e.g., `msh/US/TX` for Texas) won't work. I must ask, if only for my own curiosity, what's the point of disallowing global subscriptions but recommending subscribing to multiple topics? I hope that question doesn't come across as rude or disrespectful.

Glad to get MeshMap up and running again soon in a sanctioned manner.

**Comment by brianshea2** [comment]
Oops, my example of the US state-based subtopics was a bad one, those will still work simply by subscribing to US, of course. The issue does still apply to custom root topics of the form `msh/<custom>`, however.

**Comment by CamFlyerCH** [comment]
Hello @brianshea2 .
Yea, some of the nodes in Switzerland have switched to the msh/CH topic, and we even host a dedicated MQTT Channel there where the up- and downlink where activated. But this is almost dead now anyways with the 0-Hopping. (I still do not understand how that works). 
But to the point of the very reduced visibility of nodes on the map. All the Solar-Nodes with fixed positions configured and without a degraded postition enabled are missing.

**Comment by brianshea2** [comment]
> > Hi @garthvh! I was going to reach out to you on discord, thank you for beating me to it :)
> > Glad to hear the privacy focused improvements have been made. I'm happy to re-enable MeshMap by subscribing to each of the official region-based root topics. That will mean custom root topics won't work, however. I must ask, if only for my own curiosity, what's the point of disallowing global subscriptions but recommending subscribing to multiple topics? I hope that question doesn't come across as rude or disrespectful.
> > Glad to get MeshMap up and running again soon in a sanctioned manner.
> 
> Try this, this will allow for most custom root topic:
> 
> * msh/+/2/#
> * msh/+/+/2/#
> * msh/+/+/+/2/#
> * msh/+/+/+/+/2/#
> * msh/+/+/+/+/+/2/#

This works, but I'm not sure it's "sanctioned" (or wanted) by Meshtastic. This is basically the same as a global subscription. Then again, I'm 100% sure why global subscriptions have been disabled but then this kind of wildcard subscription still works. Would be an easy change and would let custom root topics work again. @garthvh thoughts here?

**Comment by edwinyoo44** [comment]
> > > Hi @garthvh! I was going to reach out to you on discord, thank you for beating me to it :)
> > > Glad to hear the privacy focused improvements have been made. I'm happy to re-enable MeshMap by subscribing to each of the official region-based root topics. That will mean custom root topics won't work, however. I must ask, if only for my own curiosity, what's the point of disallowing global subscriptions but recommending subscribing to multiple topics? I hope that question doesn't come across as rude or disrespectful.
> > > Glad to get MeshMap up and running again soon in a sanctioned manner.
> > 
> > 
> > Try this, this will allow for most custom root topic:
> > 
> > * msh/+/2/#
> > * msh/+/+/2/#
> > * msh/+/+/+/2/#
> > * msh/+/+/+/+/2/#
> > * msh/+/+/+/+/+/2/#
> 
> This works, but I'm not sure it's sanctioned (or wanted) by Meshtastic. This is basically the same as a global subscription. Then again, I'm not 100% sure why global subscriptions have been disabled but then this kind of wildcard subscription still works. Would be an easy change and would let custom root topics work again. @garthvh thoughts here?

My initial approach was to subscribe to country codes based on ISO 3166-1 alpha-2.
https://github.com/edwinyoo44/MeshSight/blob/cc7efb88c6dee8a90b5b0d5c77eed1244e4c31ca/MeshSight-Gateway/config_init/config.yaml#L33

**Comment by garthvh** [comment]
Having the root topic exposed is just not a best practice, while those ++ subs may work for a bit, expect that auth and topics constrained within the regions is where things are going.

**Comment by edwinyoo44** [comment]
> Having the root topic exposed is just not a best practice, while those ++ subs may work for a bit, expect that auth and topics constrained within the regions is where things are going.

I completely agree with your perspective; disabling the root topic can resolve most of the concerns. 

However, if the local community is highly concerned about privacy, using unconventional subtopics is like burying one's head in the sand—it doesn't genuinely solve the problem.
